### PR TITLE
Fix: Mixpanel

### PR DESCRIPTION
**Pull request by prabrishac**

### CODE REVIEW

**Comment by prabrishac** [comment]
UTs are yet to be fixed. Once, the review is done, will fix the UTs.

File: v0/destinations/mp/data/MPEventPropertiesConfig.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
+[
+  {
+    "sourceKeys": "request_ip",
```
Comment (arnab-p) [inline v0/destinations/mp/data/MPEventPropertiesConfig.json:3]: Should we give precedence to `context.ip` over `request_ip`?

File: v0/destinations/mp/transform.js [hunk 2]
```diff
@@ -1,21 +1,68 @@
...
 const { ConfigCategory, mappingConfig } = require("./config");
 
 const mPIdentifyConfigJson = mappingConfig[ConfigCategory.IDENTIFY.name];
+const mPProfileAndroidConfigJson =
+  mappingConfig[ConfigCategory.PROFILE_ANDROID.name];
+const mPProfileIosConfigJson = mappingConfig[ConfigCategory.PROFILE_IOS.name];
+const mPEventPropertiesConfigJson =
+  mappingConfig[ConfigCategory.EVENT_PROPERTIES.name];
+
+function getGroupKeys(config) {
```
Comment (arnab-p) [inline v0/destinations/mp/transform.js:26]: Can it be converted to a generic method and move to utils?

File: v0/destinations/mp/transform.js [hunk 3]
```diff
@@ -1,21 +1,68 @@
...
+      groupKey = groupKeyObj["groupKey"];
+      if (groupKey) {
+        groupKeys.push(groupKey);
+      }
+    });
+  }
+  return groupKeys;
+}
 
 function getEventTime(message) {
```
Comment (arnab-p) [inline v0/destinations/mp/transform.js:45]: Move to utils?

File: v0/destinations/mp/transform.js [hunk 4]
```diff
@@ -1,21 +1,68 @@
...
-  return new Date(message.originalTimestamp).toISOString();
+  return new Date(message.timestamp).toISOString();
+}
+
+function toUnixTimestamp(timestamp) {
```
Comment (arnab-p) [inline v0/destinations/mp/transform.js:49]: Move to utils?

File: v0/destinations/mp/transform.js [hunk 5]
```diff
@@ -1,21 +1,68 @@
...
+  const date = new Date(timestamp);
+  const unixTimestamp = Math.floor(date.getTime() / 1000);
+  return unixTimestamp;
+}
+
+function getTimeDifference(message) {
```
Comment (arnab-p) [inline v0/destinations/mp/transform.js:55]: Move to utils?

File: v0/destinations/mp/transform.js [hunk 6]
```diff
@@ -118,20 +199,31 @@ function getTransformedJSON(message, mappingJson) {
...
+    properties.$browser_version = browser.version;
+  }
+  const unixTimestamp = toUnixTimestamp(message.timestamp);
 
   const parameters = {
     $set: properties,
     $token: destination.Config.token,
-    $distinct_id: message.userId || message.anonymousId
+    $distinct_id: message.userId || message.anonymousId,
+    $ip: message.request_ip,
```
Comment (arnab-p) [inline v0/destinations/mp/transform.js:225]: Precedence to `context.ip`?

**Review by arnab-p — CHANGES_REQUESTED** [review]

**Review by arnab-p — CHANGES_REQUESTED** [review]
Minor comments

**Review by arnab-p — APPROVED** [review]

**Comment by mderazon** [comment]
Is there any background for this change ?
Due to this, Rudder is now sending the group id inside an array which appears in Mixpanel like this:
image

I couldn't find any reference in Mixpanel docs that they support this kind of functionality, I saw this question:
https://community.mixpanel.com/sending-data-to-mixpanel-11/single-user-part-of-multiple-groups-5991

But this kind of messes up with Mixpanel's internal group feature that they have which requires each event to have the groupKey sent as an event property as well

**Pull request merged**